### PR TITLE
[DOC] Doc for some #ceil and #floor

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Completes #ceil and #floor for Integer, Float, and Numeric.

### CODE REVIEW

File: numeric.c [hunk 1]
```diff
@@ -2170,36 +2170,82 @@ flo_ndigits(int argc, VALUE *argv)
...
+ *  |      -1 |          10 |                     -12350 |
+ *  |      -2 |         100 |                     -12400 |
+ *  |      -3 |        1000 |                     -13000 |
+ *  |      -4 |       10000 |                     -20000 |
+ *  |      -5 |      100000 |                    -100000 |
+ *  |      -6 |     1000000 |                   -1000000 |
+ *  |      -7 |    10000000 |                  -10000000 |
+ *  |      -8 |   100000000 |                 -100000000 |
+ *  |      -9 |  1000000000 |                -1000000000 |
+ *  |     -10 | 10000000000 |                          0 |
```
Comment (peterzhu2118) [inline numeric.c:2241]: I'm not sure why the result is 0 here, is it not `-10000000000`?

```ruby
-12345.6789.floor(-10) # => -10000000000
```

Comment (BurdetteLamar) [reply numeric.c:2241]: Not on my machine:
```
-12345.6789.floor(-10)
# => 0
-12345.6789.floor(-20)
# => 0
```

```
irb --version
irb 1.11.0 (2023-12-19)
```

Comment (peterzhu2118) [reply numeric.c:2241]: That's odd. What's your Ruby version (can you paste the output of `ruby -v`)? Also, what's the output of your `-12345.floor(-10)`? It should be `-10000000000`.

Comment (BurdetteLamar) [reply numeric.c:2241]:
```
ruby -v
ruby 3.3.0 (2023-12-25 revision 5124f9ac75) [x64-mingw-ucrt]
```

```
irb
-12345.floor(-10)
# => 0
```

Comment (BurdetteLamar) [reply numeric.c:2241]: I also see the difference between the two.  I'll be happy to confirm and put up the Linux results if that seems best.

Comment (peterzhu2118) [reply numeric.c:2241]: Oh I see what's going on. When the rounding will overflow, it returns 0 instead. Windows' long is 4 bytes whereas on macOS/Linux it's 8 bytes. I can reproduce the result with a larger magnitude `ndigits`.

```ruby
-12345.6789.floor(-100) # => 0
-12345.floor(-100) # => 0
12344.ceil(-100) # => 0
```

I'm not sure if this is intended behaviour. If it's not intended behaviour, then we shouldn't document it. If it is, then we should have a separate section to describe it.

@nobu do you know if this is intended behaviour?

Comment (BurdetteLamar) [reply numeric.c]: I have removed the questionable lines from the table.

**Comment by BurdetteLamar** [comment]
Had an incomplete commit for Numeric#floor, now fixed.

**Review by jeremyevans — APPROVED** [review]

File: numeric.c (hunk 1 above)
Comment (peterzhu2118) [reply numeric.c]: I opened a PR that will fix this issue in https://github.com/ruby/ruby/pull/11257.

**Review by peterzhu2118 — APPROVED** [review]

**Pull request merged**